src/app/dataset-view/additional-components/dataset-settings-component/tabs/webhooks/components/list-webhooks/list-webhooks.component.ts

```diff
@@ -0,0 +1,214 @@
+        MatIconModule,
+        MatTableModule,
+
+        //-----//
+        FeatureFlagDirective,
+    ],
+    templateUrl: "./list-webhooks.component.html",
+    styleUrls: ["./list-webhooks.component.scss"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ListWebhooksComponent implements OnInit {
+    @Input(RoutingResolvers.DATASET_SETTINGS_WEBHOOKS_KEY) public webhooksViewData: DatasetSettingsWebhookTabData;
+    public readonly DISPLAY_COLUMNS: string[] = ["event", "status", "actions"];
+    public readonly WebhookSubscriptionStatus: typeof WebhookSubscriptionStatus = WebhookSubscriptionStatus;
+    private readonly _rowsSubject$ = new BehaviorSubject<WebhookSubscription[]>([]);
+    public readonly rows$ = this._rowsSubject$.asObservable();
+    public readonly dataSource$: Observable<WebhookSubscription[]> = this.rows$;
+    private datasetWebhooksService = inject(DatasetWebhooksService);
+    private modalService = inject(ModalService);
+    private navigationService = inject(NavigationService);
+
+    public ngOnInit(): void {
+        this.fetchTableData();
+    }
+
+    private fetchTableData(): void {
+        this._rowsSubject$.next(this.webhooksViewData.subscriptions);
+    }
+
+    public get datasetBasics(): DatasetBasicsFragment {
+        return this.webhooksViewData.datasetBasics;
+    }
+
+    public createWebhook(): void {
+        this.navigationService.navigateToWebhooks({
+            accountName: this.datasetBasics.owner.accountName,
+            datasetName: this.datasetBasics.name,
+            tab: ProjectLinks.URL_WEBHOOK_NEW,
+        });
+    }
+
+    public removeWebhook(subscriptionId: string): void {
+        promiseWithCatch(
+            this.modalService.error({
+                title: "Remove webhook subscription",
+                message: `Do you want to remove webhook subscription?`,
+                bigTextBlock: "Ok",
+                noButtonText: "Cancel",
+                yesButtonText: "Ok",
+                handler: (ok) => {
+                    if (ok) {
+                        this.datasetWebhooksService
+                            .datasetWebhookRemoveSubscription(this.datasetBasics.id, subscriptionId)
+                            .pipe(take(1))
+                            .subscribe((result: boolean) => {
+                                if (result) {
+                                    const currentRows: WebhookSubscription[] = this._rowsSubject$.getValue();
+                                    const updatedRows: WebhookSubscription[] = currentRows.filter(
+                                        (row) => row.id !== subscriptionId,
+                                    );
+                                    this._rowsSubject$.next(updatedRows);
+                                }
+                            });
+                    }
+                },
+            }),
+        );
+    }
+    public pauseWebhook(subscriptionId: string): void {
```

Comment:
I think we need to find a linting rule that does not allow functions without blank lines around.